sdks/java/harness/src/main/java/org/apache/beam/fn/harness/state/FnApiStateAccessor.java

```diff
@@ -124,12 +124,14 @@ public FnApiStateAccessor(
     return new Supplier<ResultT>() {
       private ArgT memoizedArg;
       private ResultT memoizedResult;
+      private boolean initialized;
 
       @Override
       public ResultT get() {
         ArgT currentArg = arg.get();
-        if (currentArg != memoizedArg) {
+        if (currentArg != memoizedArg || (currentArg == null && !initialized)) {
```

Comment:
The initialized will become true and will stay true so checking `currentArg == null` won't do anything.
```suggestion
        if (currentArg != memoizedArg || !initialized) {
```